platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/junit/result/TestEventReporterAsListener.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing.junit.result;
+
+import org.gradle.api.internal.tasks.testing.GroupTestEventReporterInternal;
+import org.gradle.api.internal.tasks.testing.TestCompleteEvent;
+import org.gradle.api.internal.tasks.testing.TestDescriptorInternal;
+import org.gradle.api.internal.tasks.testing.TestEventReporterInternal;
+import org.gradle.api.internal.tasks.testing.TestStartEvent;
+import org.gradle.api.internal.tasks.testing.results.TestListenerInternal;
+import org.gradle.api.tasks.testing.TestEventReporter;
+import org.gradle.api.tasks.testing.TestMetadataEvent;
+import org.gradle.api.tasks.testing.TestOutputEvent;
+import org.gradle.api.tasks.testing.TestResult;
+import org.gradle.internal.exceptions.DefaultMultiCauseException;
+import org.jspecify.annotations.NullMarked;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+@NullMarked
+public final class TestEventReporterAsListener implements TestListenerInternal, AutoCloseable {
```

Comment:
Can we add some javadoc that explains what this does (it appears to be a bridge between TestListenerInternal to TestEventReporter infra)